packages/dx-react-chart/src/templates/series/path.tsx

```diff
@@ -1,28 +1,84 @@
 import * as React from 'react';
 import {
-  getAreaAnimationStyle, HOVERED, SELECTED,
+  processLineAnimation, HOVERED, SELECTED, isArrayValuesChanged, getStartCoordinates,
+  PathComponentPathState, PathPoints, Animation, isScalesChanged, getStartVal,
 } from '@devexpress/dx-chart-core';
 import { withStates } from '../../utils/with-states';
 import { PathComponentPathProps } from '../../types';
 
-class RawPath extends React.PureComponent<PathComponentPathProps> {
+class RawPath extends React.PureComponent<PathComponentPathProps, PathComponentPathState> {
+  animate: Animation | undefined;
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      coordinates: [],
+      style: undefined,
+    };
+    this.setAttribute = this.setAttribute.bind(this);
+  }
+
+  setAttribute({ coordinates, style }: { coordinates: PathPoints, style?: React.CSSProperties }) {
+    this.setState({ coordinates, style });
+  }
+
+  componentDidMount() {
+    const {
+      coordinates, animation, scales,
+    } = this.props;
+    if (!animation) {
+      this.setAttribute({ coordinates });
+    } else {
+      this.animate = animation(
+        getStartCoordinates(scales, coordinates), coordinates,
+        processLineAnimation, this.setAttribute,
+      );
+    }
+  }
+
+  componentDidUpdate({
```

Comment:
Can we make a HOC/(simple component) to reduce common code from area, bar, path, point, slice, line components?

I mean try to extract componentDidMount, ...DidUpdate, ...WillUnmount